sklearn/_loss/loss.py

```diff
@@ -725,6 +727,11 @@ class HalfTweedieLoss(BaseLoss):
     """
 
     def __init__(self, sample_weight=None, power=1.5):
+        if not isinstance(power, numbers.Number) or not math.isfinite(power):
```

Comment:
`check_scalar` is not smart enough here with the `inf` and the inclusion. I assume this is just a use case where `include_boundaries="right"` with `max_val=np.inf` means including finite vs infinite bound.

Reply:
In this case, I think we are rejecting `inf`, which means this works:

```python
from sklearn.utils.validation import check_scalar
from numbers import Real

check_scalar(np.inf, "power", target_type=Real, include_boundaries="neither",
             min_val=-np.inf, max_val=np.inf)
```